lib/src/mobile_scanner.dart

```diff
@@ -257,19 +259,21 @@ class _MobileScannerState extends State<MobileScanner>
   StreamSubscription? _subscription;
 
   Future<void> initMobileScanner() async {
-    // TODO: This will be fixed in another PR
     // If debug mode is enabled, stop the controller first before starting it.
     // If a hot-restart is initiated, the controller won't be stopped, and because
     // there is no way of knowing if a hot-restart has happened, we must assume
     // every start is a hot-restart.
-    // if (kDebugMode) {
-    //   try {
-    //     await controller.stop();
-    //   } catch (e) {
-    //     // Don't do anything if the controller is already stopped.
-    //     debugPrint('$e');
-    //   }
-    // }
+    if (kDebugMode) {
+      final platformInterface = MobileScannerPlatform.instance;
+      if (platformInterface is MethodChannelMobileScanner) {
```

Comment:
Can we clean this up with an if-case? Just a suggestion though.

```suggestion
      if (MobileScannerPlatform.instance case final MethodChannelMobileScanner implementation) {
```